cw/simple_capture_aes_sca.py

```diff
@@ -78,121 +107,160 @@ def abort_handler_during_loop(project, sig, frame):
         file_name_local = f"{project_name}_data/scope_config_{now_str}.lss"
         waverunner.save_setup_to_local_file(file_name_local)
 
-    # Check if batch mode is requested TODO: not supported yet
-    if cfg["waverunner"]["num_segments"] > 1:
-        print("Warning: Sequence (batch) mode not supported yet")
-    if cfg["cwfpgahusky"]["num_segments"] > 1:
-        print("Warning: Sequence (batch) mode not supported yet")
+    # Preparation of Key and plaintext generation ------------------------------
 
-    # Register ctrl-c handler to store traces on abort
-    signal.signal(signal.SIGINT, partial(abort_handler_during_loop, project))
+    # This test currently uses one fixed key and generates random texts through
+    # AES encryption using a generation key. The batch mode uses ciphertexts as
+    # next text input here and in the OT device SW.
+
+    # Load fixed key and initial text values from cfg
+    key_fixed = bytearray(cfg["test"]["key_fixed"])
+    print(f'Using key: {binascii.b2a_hex(bytes(key_fixed))}')
+    text = bytearray(cfg["test"]["text_fixed"])
+
+    # Cipher to compute expected responses
+    cipher = AES.new(bytes(key_fixed), AES.MODE_ECB)
 
-    # Preparation of Key and plaintext generation
-    # Generate key at random based on test_random_seed (not used atm)
-    random.seed(cfg["test"]["test_random_seed"])
-    key = bytearray(cfg["test"]["key_len_bytes"])
-    for i in range(0, cfg["test"]["key_len_bytes"]):
-        key[i] = random.randint(0, 255)
-    # Load initial key and text values from cfg
-    key = bytearray(cfg["test"]["key"])
-    print(f'Using key: {binascii.b2a_hex(bytes(key))}')
-    text = bytearray(cfg["test"]["text"])
     # Prepare generation of new texts/keys by encryption using key_for_generation
     key_for_gen = bytearray(cfg["test"]["key_for_gen"])
     cipher_gen = AES.new(bytes(key_for_gen), AES.MODE_ECB)
 
+    # Seed the target's PRNGs for masking, turns off masking when '0'
+    cwfpgahusky.target.simpleserial_write("l", cfg["test"]["lfsr_seed"].to_bytes(4, "little"))
```

Comment:
Just to clarify, this seeds the SW PRNG that is only used for initial key sharing.
This is unrelated to the internal HW PRNG used for remasking inside the datapath.

To avoid confusion I would change this comment to something like `#Seed the target's PRNGs for initial key sharing.`

Reply:
Thanks! 
This routine also does `transaction.force_masks = true;` on the device `(aes_serial.c)`
Maybe something like:
# Seed the target's PRNGs for initial key masking, and turn off masking when '0'

@vogelpi did I get it right how I "turn off the masks"?

Reply:
fixed